### PR TITLE
Layermanager, basemap gallery & toolbar updates

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

 Started as an UI update for toolbar and basemap gallery ....ended up with a bit more.
- split layer editor into two main parts - settings and sublayers. The parts are indenepndent so it is now possible to open each for differnet layer
- both parts are created where right where it is needed eg. we dont have to move it around the DOM anymore
- refactored sublayer visibiltiy management in editor
- simplify basemap gallery - chose active basemap or open layereditor in dialog.
- simplify draw-toolbar. It now opens in popup window with content of draw-panel instead of previous rather complicated toolbar tool which did exactly the same as panel just different
- some UI updates for toolbar and basemap gallery
- transformed componetns into standalone along the way

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Comment by jmacura** [comment]
solves #5308

**Comment by jmacura** [comment]
Wow, such nice! Much upgrade 🥇 

image
image

**Review by jmacura — CHANGES_REQUESTED** [review]
Somehow I made renaming a baselayer via dialog-box to fail, but cannot replicate.
image

**Comment by jmacura** [comment]
If I open layer-editor from the basemap gallery, then move the baselayer into a folder, then close the layer-editor window. A multitude of errors appear in the console and the layer-editor won't open again.
image
image

**Comment by jmacura** [comment]
One tiny UX issue is that if I rename the layer via layer-editor in dialog window, the new layer title is not reflected in the window title when saved (the window must be closed and reopened).
image
image

File: projects/hslayers/components/draw/draw-panel/draw-panel.component.ts [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
 @Component({
   selector: 'hs-draw-panel',
   templateUrl: './draw-panel.component.html',
   styleUrls: ['./draw-panel.component.scss'],
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    TranslateCustomPipe,
+    NgClass,
+    NgForOf,
```
Comment (jmacura) [inline projects/hslayers/components/draw/draw-panel/draw-panel.component.ts:32]: Shouldn't be necessary when only control flow blocks are used (no `*ngFor`).

Comment (FilipLeitner) [reply projects/hslayers/components/draw/draw-panel/draw-panel.component.ts]: [9335610](https://github.com/hslayers/hslayers-ng/pull/5317/commits/9335610ba3ac7ace4a16b9fad6b214eb3541acad)

**Comment by FilipLeitner** [comment]
> One tiny UX issue is that if I rename the layer via layer-editor in dialog window, the new layer title is not reflected in the window title when saved (the window must be closed and reopened).

Should be solved by [806c36a](https://github.com/hslayers/hslayers-ng/pull/5317/commits/806c36a4484ebdf467dd71f4cf775db0a26de588)

>If I open layer-editor from the basemap gallery, then move the baselayer into a folder, then close the layer-editor window and open the LayerManager panel. A multitude of errors appear in the console and the layer-editor won't open again.

Should be solved by [e1a885c](https://github.com/hslayers/hslayers-ng/pull/5317/commits/e1a885c2d8159b031e5035e282ec887b6c050430)

> An error. Similar to the one below. Steps to reproduce:
open LayerManager panel
open layer-editor from basemap-gallery.
rename layer (via layer-editor in dialog window)
see errors in console
try to rename the layer again (via layer-editor in dialog window)

This one I could not reporduce

**Comment by jmacura** [comment]
> > An error. Similar to the one below. Steps to reproduce:
> > open LayerManager panel
> > open layer-editor from basemap-gallery.
> > rename layer (via layer-editor in dialog window)
> > see errors in console
> > try to rename the layer again (via layer-editor in dialog window)
> 
> This one I could not reporduce

I can't replicate it with the latest commits either.
Thanks for all the improvements!

**Comment by jmacura** [comment]
Sometimes, I get a toolbar with: search, draw, measure tools and sometimes with a reversed order. Occasionally even with the search in the middle (but it's rare).
Do you observe it too?
It is likely unrelated to your current changes, as I observe it also on develop. Perhaps I gonna spin it out to a new issue.

obrazek
obrazek
obrazek

**Comment by FilipLeitner** [comment]
> Sometimes, I get a toolbar with: search, draw, measure tools and sometimes with a reversed order. Occasionally even with the search in the middle (but it's rare). Do you observe it too? It is likely unrelated to your current changes, as I observe it also on develop. Perhaps I gonna spin it out to a new issue.
> 

Yup I have seen it. It is most likely a result of the variablity in these lazy imports. 
https://github.com/hslayers/hslayers-ng/blob/7b4ffe70535a55e3e25c8752718d847e025026d2/projects/hslayers/services/panel-constructor/overlay-constructor.service.ts#L37-L48

**Pull request merged**